plugins/vppv2/ipsecplugin/vppcalls/ipsec_vppcalls.go

```diff
@@ -0,0 +1,228 @@
+			req.RemoteAddressStop = net.ParseIP(spd.RemoteAddrStop).To4()
+			req.LocalAddressStart = net.ParseIP(spd.LocalAddrStart).To4()
+			req.LocalAddressStop = net.ParseIP(spd.LocalAddrStop).To4()
+		}
+	} else {
+		req.RemoteAddressStart = net.ParseIP("0.0.0.0").To4()
+		req.RemoteAddressStop = net.ParseIP("255.255.255.255").To4()
+		req.LocalAddressStart = net.ParseIP("0.0.0.0").To4()
+		req.LocalAddressStop = net.ParseIP("255.255.255.255").To4()
+	}
+	reply := &api.IpsecSpdAddDelEntryReply{}
+
+	if err := h.callsChannel.SendRequest(req).ReceiveReply(reply); err != nil {
+		return err
+	} else if reply.Retval != 0 {
+		return fmt.Errorf("%s returned %d", reply.GetMessageName(), reply.Retval)
+	}
+
+	return nil
+}
+
+func (h *IPSecVppHandler) interfaceAddDelSpd(spdID, swIfIdx uint32, isAdd bool) error {
+	req := &api.IpsecInterfaceAddDelSpd{
+		IsAdd:     boolToUint(isAdd),
+		SwIfIndex: swIfIdx,
+		SpdID:     spdID,
+	}
+	reply := &api.IpsecInterfaceAddDelSpdReply{}
+
+	if err := h.callsChannel.SendRequest(req).ReceiveReply(reply); err != nil {
+		return err
+	} else if reply.Retval != 0 {
+		return fmt.Errorf("%s returned %d", reply.GetMessageName(), reply.Retval)
+	}
+
+	return nil
+}
+
+func (h *IPSecVppHandler) sadAddDelEntry(sa *ipsec.SecurityAssociation, isAdd bool) error {
+	cryptoKey, err := hex.DecodeString(sa.CryptoKey)
+	if err != nil {
+		return err
+	}
+	integKey, err := hex.DecodeString(sa.IntegKey)
+	if err != nil {
+		return err
+	}
+
+	saID, err := strconv.Atoi(sa.Index)
+	if err != nil {
+		return err
+	}
+
+	req := &api.IpsecSadAddDelEntry{
+		IsAdd:                     boolToUint(isAdd),
+		SadID:                     uint32(saID),
+		Spi:                       sa.Spi,
+		Protocol:                  uint8(sa.Protocol),
+		CryptoAlgorithm:           uint8(sa.CryptoAlg),
+		CryptoKey:                 cryptoKey,
+		CryptoKeyLength:           uint8(len(cryptoKey)),
+		IntegrityAlgorithm:        uint8(sa.IntegAlg),
+		IntegrityKey:              integKey,
+		IntegrityKeyLength:        uint8(len(integKey)),
+		UseExtendedSequenceNumber: boolToUint(sa.UseEsn),
+		UseAntiReplay:             boolToUint(sa.UseAntiReplay),
+		UDPEncap:                  boolToUint(sa.EnableUdpEncap),
+	}
+	if sa.TunnelSrcAddr != "" {
+		req.IsTunnel = 1
+		isIPv6, err := addrs.IsIPv6(sa.TunnelSrcAddr)
+		if err != nil {
+			return err
+		}
+		if isIPv6 {
+			req.IsTunnelIPv6 = 1
+			req.TunnelSrcAddress = net.ParseIP(sa.TunnelSrcAddr).To16()
+			req.TunnelDstAddress = net.ParseIP(sa.TunnelDstAddr).To16()
+		} else {
+			req.IsTunnelIPv6 = 0
+			req.TunnelSrcAddress = net.ParseIP(sa.TunnelSrcAddr).To4()
+			req.TunnelDstAddress = net.ParseIP(sa.TunnelDstAddr).To4()
+		}
+	}
+	reply := &api.IpsecSadAddDelEntryReply{}
+
+	if err := h.callsChannel.SendRequest(req).ReceiveReply(reply); err != nil {
+		return err
+	} else if reply.Retval != 0 {
+		return fmt.Errorf("%s returned %d", reply.GetMessageName(), reply.Retval)
```

Comment:
This is not needed anymore and was already removed from other vppcalls packages.